docs/tutorials/multi-d-datasets.ipynb

```diff
@@ -259,7 +261,7 @@
     "offset = d.copy()\n",
```

Comment:
Tried adding a comment above, but this is the closest I can get.

```Python
try:
    d = sc.concatenate(d, d, Dim.X)
except RuntimeError:
    print("Failed as expected!")
```
a couple of cells above is not behaving correctly. The `concatenate` operation does not fail.
However, nothing seems to happen at all.

Dataset `d` before:
```
<scipp.Dataset>
Dimensions: {{Dim.X, 8}, {Dim.Y, 10}, {Dim.Z, 8}}
Coordinates:
    Dim.Z                     double    [m]              (Dim.Z [bin-edges])  [1.000000, 2.000000, 3.000000, 4.000000, ..., 6.000000, 7.000000, 8.000000, 9.000000]
    Dim.X                     double    [m]              (Dim.X [bin-edges])  [1.000000, 2.000000, 3.000000, 4.000000, ..., 6.000000, 7.000000, 8.000000, 9.000000]
    Dim.Y                     double    [m]              (Dim.Y [bin-edges])  [0.000000, 1.000000, 2.000000, 3.000000, ..., 7.000000, 8.000000, 9.000000, 10.000000]
Labels:
Data:
    alice                     double    [dimensionless]  (Dim.Z, Dim.Y, Dim.X)  [-0.116024, 0.007157, -0.558373, -0.812718, ..., 0.247166, -0.357067, 0.178212, 0.287054]  [0.265719, 0.211498, 0.135339, 0.227581, ..., 0.266205, 0.164554, 0.196404, 0.203193]
    bob                       double    [dimensionless]  (Dim.X, Dim.Z)  [1.100000, 1.200000, 1.300000, 1.400000, ..., 8.500000, 8.600000, 8.700000, 8.800000]  [0.072206, 0.015697, 0.097417, 0.073549, ..., 0.082262, 0.080639, 0.014295, 0.052393]
Attributes:
```
and after:
```
<scipp.Dataset>
Dimensions: {{Dim.Y, 10}, {Dim.Z, 8}, {Dim.X, 8}}
Coordinates:
    Dim.X                     double    [m]              (Dim.X [bin-edges])  [1.000000, 2.000000, 3.000000, 4.000000, ..., 6.000000, 7.000000, 8.000000, 9.000000]
    Dim.Z                     double    [m]              (Dim.Z [bin-edges])  [1.000000, 2.000000, 3.000000, 4.000000, ..., 6.000000, 7.000000, 8.000000, 9.000000]
    Dim.Y                     double    [m]              (Dim.Y [bin-edges])  [0.000000, 1.000000, 2.000000, 3.000000, ..., 7.000000, 8.000000, 9.000000, 10.000000]
Labels:
Data:
    alice                     double    [dimensionless]  (Dim.Z, Dim.Y, Dim.X)  [-0.116024, 0.007157, -0.558373, -0.812718, ..., 0.247166, -0.357067, 0.178212, 0.287054]  [0.265719, 0.211498, 0.135339, 0.227581, ..., 0.266205, 0.164554, 0.196404, 0.203193]
    bob                       double    [dimensionless]  (Dim.X, Dim.Z)  [1.100000, 1.200000, 1.300000, 1.400000, ..., 8.500000, 8.600000, 8.700000, 8.800000]  [0.072206, 0.015697, 0.097417, 0.073549, ..., 0.082262, 0.080639, 0.014295, 0.052393]
Attributes:
```

Reply:
Shouldn't it be `{Dim.X, 15}` if it had not failed? Instead it is `8`, as before?

Reply:
yes, nothing appears to happen

Reply:
... so I do not understand why you say above *it does not fail*?

Reply:
I removed the `try` and ran
```Python
d = sc.concatenate(d, d, Dim.X)
d
```
and this is what i'm showing above

Reply:
no errors are printed

Reply:
So it does not fail but returns an unchanged dataset?

Reply:
Ohhh, I see, this is due to a bad special-case handling for identical inputs. Will fix it.

Reply:
#509